### ISSUE TITLE
Is the ANARIMemoryDeleter called when managed arrays are destructed?

### ISSUE THREAD

**Issue by johguenther** [KhronosGroup/ANARI-Docs#19]
I think this is currently undefined behavior. The call is not needed (because the array is managed and owned by ANARI). Is it then an error to create a managed array with the callback set (`appMemory=NULL` but `ANARIMemoryDeleter != NULL`)?

**Comment by jeffamstutz** [comment]
You are correct, the simplest description should be "if `appMemory=NULL` then only the type and size parameters are used".
